modules/tensor_mechanics/src/materials/FiniteStrainCPSlipRateRes.C

```diff
@@ -0,0 +1,368 @@
+
+  eqv_slip_incr = iden - eqv_slip_incr;
+
+  _fp_inv = _fp_old_inv * eqv_slip_incr;
+  _fe = _dfgrd_tmp * _fp_inv;
+
+  ce = _fe.transpose() * _fe;
+  ee = ce - iden;
+  ee *= 0.5;
+
+  _pk2_tmp = _elasticity_tensor[_qp] * ee;
+
+  for (unsigned int i = 0; i < _nss; ++i)
+    _tau[i] = _pk2_tmp.doubleContraction(_s0[i]);
+
+  update_slip_system_resistance();
+  getSlipIncrements();
+
+  if (_err_tol)
+    return;
+
+  for (unsigned int i = 0; i < _nss; ++i)
+    _resid[i] = _slip_rate[i] - _slip_incr[i] / _dt;
+}
+
+void
+FiniteStrainCPSlipRateRes::calcJacobianSlipRate()
+{
+  //_dsliprate_dsliprate not reinitialized to zero, hence order is important
+  calcDtauDsliprate();
+  calcDgssDsliprate();
+
+  for (unsigned int i = 0; i < _nss; ++i)
+    for (unsigned int j = 0; j < _nss; ++j)
+    {
+      _jacob(i,j) = 0.0;
+      if (i == j) _jacob(i,j) += 1.0;
+      _jacob(i,j) -= _dsliprate_dsliprate(i,j);
+    }
+}
+
+void
+FiniteStrainCPSlipRateRes::calcDtauDsliprate()
+{
+  RankFourTensor dfedfpinv, deedfe, dfpinvdpk2;
+  std::vector<RankTwoTensor> dtaudpk2(_nss), dfpinvdsliprate(_nss);
+
+  for (unsigned int i = 0; i < _nss; ++i)
+  {
+    dtaudpk2[i] = _s0[i];
+    dfpinvdsliprate[i] = - _fp_old_inv * _s0[i] * _dt;
+  }
+
+  for (unsigned int i = 0; i < LIBMESH_DIM; ++i)
+    for (unsigned int j = 0; j < LIBMESH_DIM; ++j)
+      for (unsigned int k = 0; k < LIBMESH_DIM; ++k)
+        dfedfpinv(i,j,k,j) = _dfgrd_tmp(i,k);
+
+  for (unsigned int i = 0; i < LIBMESH_DIM; ++i)
+    for (unsigned int j = 0; j < LIBMESH_DIM; ++j)
+      for (unsigned int k = 0; k < LIBMESH_DIM; ++k)
+      {
+        deedfe(i,j,k,i) = deedfe(i,j,k,i) + _fe(k,j) * 0.5;
+        deedfe(i,j,k,j) = deedfe(i,j,k,j) + _fe(k,i) * 0.5;
+      }
+
+  RankFourTensor dpk2dfpinv;
+
+  dpk2dfpinv = _elasticity_tensor[_qp] * deedfe * dfedfpinv;
+
+  for (unsigned int i = 0; i < _nss; ++i)
+    for (unsigned int j = 0; j < _nss; ++j)
+      _dsliprate_dsliprate(i,j) = _dslipdtau[i] * dtaudpk2[i].doubleContraction(dpk2dfpinv * dfpinvdsliprate[j]);
+}
+
+void
+FiniteStrainCPSlipRateRes::calcDgssDsliprate()
+{
+  for (unsigned int i = 0; i < _nss; ++i)
+    for (unsigned int j = 0; j < _nss; ++j)
+      _dsliprate_dsliprate(i,j) += _dsliprate_dgss[i] * _dgss_dsliprate[i][j];
+}
+
+void
+FiniteStrainCPSlipRateRes::getSlipIncrements()
+{
+  FiniteStrainCrystalPlasticity::getSlipIncrements();
+
+  if (_err_tol)
+    return;
+
+  for (unsigned int i = 0; i < _nss; ++i)
+    _dslipdtau[i] /= _dt;
+
+  for (unsigned int i = 0; i < _nss; ++i)
+    _dsliprate_dgss[i] = - _a0[i] / _xm[i] * std::pow(std::abs(_tau[i] / _gss_tmp[i]), 1.0 / _xm[i] - 1.0 ) * _tau[i] / std::pow(_gss_tmp[i], 2.0);
+}
+
+void
+FiniteStrainCPSlipRateRes::calcUpdate()
+{
+  DenseMatrix<Real> A(_nss, _nss);
+  DenseVector<Real> r(_nss);
+  DenseVector<Real> x(_nss);
+
+  for (unsigned int i = 0; i < _nss; ++i)
+    for (unsigned int j = 0; j < _nss; ++j)
+      A(i,j) = _jacob(i,j);
```

Comment:
You can do a simple assignment. That was the point of making _jacob a DenseMatrix.